pkg/image/containerdregistry/resolver.go

```diff
@@ -85,14 +87,31 @@ func credential(cfg *configfile.ConfigFile) func(string) (string, string, error)
 	}
 }
 
-func loadConfig(dir string) (*configfile.ConfigFile, error) {
+func loadConfig(dir string) (cfg *configfile.ConfigFile, err error) {
 	if dir == "" {
 		dir = config.Dir()
 	}
 
-	cfg, err := config.Load(dir)
-	if err != nil {
-		return nil, err
+	dockerConfigJSON := filepath.Join(dir, config.ConfigFileName)
+	cfg = configfile.New(dockerConfigJSON)
+
+	switch _, err := os.Stat(dockerConfigJSON); {
+	case err == nil:
+		cfg, err = config.Load(dir)
+		if err != nil {
+			return cfg, err
+		}
+	case os.IsNotExist(err):
+		podmanConfig := filepath.Join(os.Getenv("XDG_RUNTIME_DIR"), "containers/auth.json")
```

Comment:
The `XDG_RUNTIME_DIR` isn't always set in the environment right? And in nearly all cases, Linux users use the default XDG directories? Should we use something like https://github.com/adrg/xdg?

Reply:
Nice, thanks for sharing this library. I will definitely incorporate this to ensure cross-platform compatibility.